File: parliament_assets/theme/Parliament1812Theme.swift
// Parliament1812Theme.swift
// 1812 國會風雲 - iOS Theme Configuration
// Based on Figma Design System

import SwiftUI

// MARK: - Color Palette
extension Color {
    struct Parliament1812 {
        // Primary Colors
        static let darkNavy = Color(hex: "1a1a2e")          // PRIMARY BACKGROUND
        static let secondaryNavy = Color(hex: "16213e")      // CARD BACKGROUNDS
        static let darkBrown = Color(hex: "1a1614")          // Alternative dark background
        static let cardBrown = Color(hex: "241b14")          // Card background variant
        
        // Accent Colors
        static let antiqueGold = Color(hex: "d4af37")        // PRIMARY ACCENT
        static let saddleBrown = Color(hex: "8b4513")        // SECONDARY ACCENT
        static let mutedGold = Color(hex: "b8941f")          // Gradient gold
        
        // Text Colors
        static let parchmentCream = Color(hex: "f5e6d3")     // PRIMARY TEXT
        static let mutedText = Color(hex: "b8a07e")          // SECONDARY TEXT
        static let subtleText = Color(hex: "8b7753")         // Tertiary text
        
        // Political Party Colors
        static let toryBlue = Color(hex: "1e3a5f")           // TORY PARTY
        static let whigOrange = Color(hex: "cc7722")         // WHIG PARTY
        static let neutral = Color(hex: "8b7753")            // NEUTRAL
        
        // Voting Colors
        static let ayeGreen = Color(hex: "2d5a27")           // AYE/SUCCESS
        static let nayCrimson = Color(hex: "8b2500")         // NAY/DANGER
        
        // Wax Seal Colors
        static let sealDark = Color(hex: "6e1e00")
        static let sealMid = Color(hex: "8b2500")
        static let sealLight = Color(hex: "a3320b")
    }
}

// MARK: - Hex Color Extension
extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3:
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6:
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8:
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (255, 0, 0, 0)
        }
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

// MARK: - Typography
struct Parliament1812Typography {
    // Primary font: Georgia (serif)
    static let titleLarge = Font.custom("Georgia", size: 48).weight(.bold)
    static let titleMedium = Font.custom("Georgia", size: 32).weight(.bold)
    static let titleSmall = Font.custom("Georgia", size: 24).weight(.semibold)
    static let heading = Font.custom("Georgia", size: 20).weight(.semibold)
    static let body = Font.custom("Georgia", size: 16)
    static let caption = Font.custom("Georgia", size: 14)
    static let small = Font.custom("Georgia", size: 12)
    
    // Letter spacing
    static let wideSpacing: CGFloat = 4
    static let normalSpacing: CGFloat = 2
}

// MARK: - Button Styles
struct GoldButtonStyle: ButtonStyle {
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .padding(.horizontal, 24)
            .padding(.vertical, 14)
            .background(
                LinearGradient(
                    colors: [Color.Parliament1812.antiqueGold, Color.Parliament1812.mutedGold],
                    startPoint: .topLeading,
                    endPoint: .bottomTrailing
                )
            )
            .foregroundColor(Color.Parliament1812.darkNavy)
            .font(.custom("Georgia", size: 16).weight(.semibold))
            .clipShape(Parliament1812ButtonShape())
            .shadow(color: Color.Parliament1812.antiqueGold.opacity(0.4), radius: 8, y: 4)
            .scaleEffect(configuration.isPressed ? 0.98 : 1.0)
    }
}

struct OutlineButtonStyle: ButtonStyle {
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .padding(.horizontal, 24)
            .padding(.vertical, 14)
            .background(Color.clear)
            .foregroundColor(Color.Parliament1812.antiqueGold)
            .font(.custom("Georgia", size: 16).weight(.semibold))
            .overlay(
                Parliament1812ButtonShape()
                    .stroke(Color.Parliament1812.antiqueGold, lineWidth: 2)
            )
            .clipShape(Parliament1812ButtonShape())
            .scaleEffect(configuration.isPressed ? 0.98 : 1.0)
    }
}

struct AyeButtonStyle: ButtonStyle {
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .padding(.horizontal: 32, .vertical: 16)
            .background(Color.Parliament1812.ayeGreen)
            .foregroundColor(Color.Parliament1812.parchmentCream)
            .font(.custom("Georgia", size: 18).weight(.bold))
            .clipShape(RoundedRectangle(cornerRadius: 8))
            .scaleEffect(configuration.isPressed ? 0.95 : 1.0)
    }
}

struct NayButtonStyle: ButtonStyle {
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .padding(.horizontal: 32, .vertical: 16)
            .background(Color.Parliament1812.nayCrimson)
            .foregroundColor(Color.Parliament1812.parchmentCream)
            .font(.custom("Georgia", size: 18).weight(.bold))
            .clipShape(RoundedRectangle(cornerRadius: 8))
            .scaleEffect(configuration.isPressed ? 0.95 : 1.0)
    }
}

// MARK: - Custom Shapes
struct Parliament1812ButtonShape: Shape {
    func path(in rect: CGRect) -> Path {
        let cornerCut: CGFloat = 12
        var path = Path()
        
        path.move(to: CGPoint(x: 0, y: 0))
        path.addLine(to: CGPoint(x: rect.width, y: 0))
        path.addLine(to: CGPoint(x: rect.width, y: rect.height - cornerCut))
        path.addLine(to: CGPoint(x: rect.width - cornerCut, y: rect.height))
        path.addLine(to: CGPoint(x: 0, y: rect.height))
        path.closeSubpath()
        
        return path
    }
}

// MARK: - Card Style
struct Parliament1812Card<Content: View>: View {
    let content: Content
    
    init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }
    
    var body: some View {
        content
            .padding()
            .background(
                RoundedRectangle(cornerRadius: 12)
                    .fill(Color.Parliament1812.cardBrown.opacity(0.9))
                    .overlay(
                        RoundedRectangle(cornerRadius: 12)
                            .stroke(Color.Parliament1812.antiqueGold.opacity(0.3), lineWidth: 1)
                    )
            )
            .shadow(color: .black.opacity(0.5), radius: 10, y: 4)
    }
}

// MARK: - Input Field Style
struct Parliament1812TextField: View {
    let placeholder: String
    @Binding var text: String
    var isCode: Bool = false
    
    var body: some View {
        TextField(placeholder, text: $text)
            .padding()
            .background(Color.Parliament1812.cardBrown.opacity(0.8))
            .foregroundColor(Color.Parliament1812.parchmentCream)
            .font(isCode ? .system(size: 24, design: .monospaced) : .custom("Georgia", size: 16))
            .overlay(
                Parliament1812ButtonShape()
                    .stroke(Color.Parliament1812.antiqueGold, lineWidth: 2)
            )
            .clipShape(Parliament1812ButtonShape())
            .autocapitalization(isCode ? .allCharacters : .words)
    }
}

// MARK: - Decorative Elements
struct CornerFlourish: View {
    enum Position {
        case topLeft, topRight, bottomLeft, bottomRight
    }
    
    let position: Position
    let size: CGFloat
    let color: Color
    
    init(position: Position = .topLeft, size: CGFloat = 32, color: Color = .Parliament1812.antiqueGold) {
        self.position = position
        self.size = size
        self.color = color
    }
    
    var body: some View {
        Image(systemName: "leaf.fill")
            .resizable()
            .frame(width: size, height: size)
            .foregroundColor(color.opacity(0.4))
            .rotationEffect(rotationAngle)
            .scaleEffect(scaleEffect)
    }
    
    private var rotationAngle: Angle {
        switch position {
        case .topLeft: return .degrees(45)
        case .topRight: return .degrees(135)
        case .bottomLeft: return .degrees(-45)
        case .bottomRight: return .degrees(-135)
        }
    }
    
    private var scaleEffect: CGSize {
        switch position {
        case .topRight, .bottomRight: return CGSize(width: -1, height: 1)
        default: return CGSize(width: 1, height: 1)
        }
    }
}

struct DividerLine: View {
    var body: some View {
        HStack(spacing: 8) {
            Rectangle()
                .fill(LinearGradient(
                    colors: [.clear, Color.Parliament1812.antiqueGold],
                    startPoint: .leading,
                    endPoint: .trailing
                ))
                .frame(width: 64, height: 1)
            
            Diamond()
                .fill(Color.Parliament1812.antiqueGold)
                .frame(width: 6, height: 6)
            
            Rectangle()
                .fill(LinearGradient(
                    colors: [Color.Parliament1812.antiqueGold, .clear],
                    startPoint: .leading,
                    endPoint: .trailing
                ))
                .frame(width: 64, height: 1)
        }
        .opacity(0.6)
    }
}

struct Diamond: Shape {
    func path(in rect: CGRect) -> Path {
        var path = Path()
        path.move(to: CGPoint(x: rect.midX, y: rect.minY))
        path.addLine(to: CGPoint(x: rect.maxX, y: rect.midY))
        path.addLine(to: CGPoint(x: rect.midX, y: rect.maxY))
        path.addLine(to: CGPoint(x: rect.minX, y: rect.midY))
        path.closeSubpath()
        return path
    }
}

// MARK: - Crown Icon
struct CrownIcon: View {
    let size: CGFloat
    let color: Color
    
    init(size: CGFloat = 24, color: Color = .Parliament1812.antiqueGold) {
        self.size = size
        self.color = color
    }
    
    var body: some View {
        Image(systemName: "crown.fill")
            .resizable()
            .aspectRatio(contentMode: .fit)
            .frame(width: size, height: size)
            .foregroundColor(color)
    }
}

// MARK: - Hexagon Shape
struct HexagonShape: Shape {
    func path(in rect: CGRect) -> Path {
        let center = CGPoint(x: rect.midX, y: rect.midY)
        let radius = min(rect.width, rect.height) / 2
        var path = Path()
        
        for i in 0..<6 {
            let angle = CGFloat(i) * .pi / 3 - .pi / 2
            let point = CGPoint(
                x: center.x + radius * cos(angle),
                y: center.y + radius * sin(angle)
            )
            if i == 0 {
                path.move(to: point)
            } else {
                path.addLine(to: point)
            }
        }
        path.closeSubpath()
        return path
    }
}

// MARK: - Party Badge
struct PartyBadge: View {
    enum Party {
        case tory, whig, neutral
        
        var color: Color {
            switch self {
            case .tory: return .Parliament1812.toryBlue
            case .whig: return .Parliament1812.whigOrange
            case .neutral: return .Parliament1812.neutral
            }
        }
        
        var nameChinese: String {
            switch self {
            case .tory: return "托利黨"
            case .whig: return "輝格黨"
            case .neutral: return "中立"
            }
        }
        
        var nameEnglish: String {
            switch self {
            case .tory: return "TORY"
            case .whig: return "WHIG"
            case .neutral: return "NEUTRAL"
            }
        }
    }
    
    let party: Party
    
    var body: some View {
        HStack(spacing: 4) {
            Circle()
                .fill(party.color)
                .frame(width: 8, height: 8)
            Text(party.nameChinese)
                .font(.custom("Georgia", size: 12))
                .foregroundColor(party.color)
        }
        .padding(.horizontal, 12)
        .padding(.vertical, 6)
        .background(
            Capsule()
                .stroke(party.color, lineWidth: 1)
        )
    }
}

// MARK: - Preview
#Preview {
    ZStack {
        Color.Parliament1812.darkBrown
            .ignoresSafeArea()
        
        VStack(spacing: 24) {
            // Title
            Text("1812")
                .font(Parliament1812Typography.titleLarge)
                .foregroundColor(.Parliament1812.antiqueGold)
                .shadow(color: .Parliament1812.antiqueGold.opacity(0.4), radius: 15)
            
            Text("國會風雲")
                .font(Parliament1812Typography.titleSmall)
                .foregroundColor(.Parliament1812.mutedText)
            
            DividerLine()
            
            // Buttons
            Button("建立房間") {}
                .buttonStyle(GoldButtonStyle())
            
            Button("加入房間") {}
                .buttonStyle(OutlineButtonStyle())
            
            // Party badges
            HStack(spacing: 16) {
                PartyBadge(party: .tory)
                PartyBadge(party: .whig)
            }
            
            // Card example
            Parliament1812Card {
                VStack(alignment: .leading, spacing: 8) {
                    Text("議案表決")
                        .font(.custom("Georgia", size: 18))
                        .foregroundColor(.Parliament1812.antiqueGold)
                    Text("在攝政王的注視下...")
                        .font(.custom("Georgia", size: 14))
                        .foregroundColor(.Parliament1812.mutedText)
                }
            }
        }
        .padding()
    }
}

File: parliament_assets/theme/Parliament1812Theme.kt
// Parliament1812Theme.kt
// 1812 國會風雲 - Android Theme Configuration
// Based on Figma Design System

package com.parliament1812.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.foundation.shape.CutCornerShape
import androidx.compose.foundation.shape.GenericShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

// ============================================
// MARK: - Color Palette
// ============================================
object Parliament1812Colors {
    // Primary Colors
    val DarkNavy = Color(0xFF1A1A2E)          // PRIMARY BACKGROUND
    val SecondaryNavy = Color(0xFF16213E)     // CARD BACKGROUNDS
    val DarkBrown = Color(0xFF1A1614)         // Alternative dark background
    val CardBrown = Color(0xFF241B14)         // Card background variant
    
    // Accent Colors
    val AntiqueGold = Color(0xFFD4AF37)       // PRIMARY ACCENT
    val SaddleBrown = Color(0xFF8B4513)       // SECONDARY ACCENT
    val MutedGold = Color(0xFFB8941F)         // Gradient gold
    
    // Text Colors
    val ParchmentCream = Color(0xFFF5E6D3)    // PRIMARY TEXT
    val MutedText = Color(0xFFB8A07E)         // SECONDARY TEXT
    val SubtleText = Color(0xFF8B7753)        // Tertiary text
    
    // Political Party Colors
    val ToryBlue = Color(0xFF1E3A5F)          // TORY PARTY
    val WhigOrange = Color(0xFFCC7722)        // WHIG PARTY
    val Neutral = Color(0xFF8B7753)           // NEUTRAL
    
    // Voting Colors
    val AyeGreen = Color(0xFF2D5A27)          // AYE/SUCCESS
    val NayCrimson = Color(0xFF8B2500)        // NAY/DANGER
    
    // Wax Seal Colors
    val SealDark = Color(0xFF6E1E00)
    val SealMid = Color(0xFF8B2500)
    val SealLight = Color(0xFFA3320B)
    
    // With Alpha
    val GoldBorder = AntiqueGold.copy(alpha = 0.3f)
    val CardShadow = Color.Black.copy(alpha = 0.5f)
}

// ============================================
// MARK: - Typography
// ============================================
// Note: Add Georgia or similar serif font to res/font/
val GeorgiaFamily = FontFamily(
    Font(R.font.georgia_regular, FontWeight.Normal),
    Font(R.font.georgia_bold, FontWeight.Bold),
)

// Fallback if Georgia not available
val SerifFamily = FontFamily.Serif

object Parliament1812Typography {
    val TitleLarge = TextStyle(
        fontFamily = SerifFamily,
        fontWeight = FontWeight.Bold,
        fontSize = 48.sp,
        letterSpacing = 8.sp
    )
    
    val TitleMedium = TextStyle(
        fontFamily = SerifFamily,
        fontWeight = FontWeight.Bold,
        fontSize = 32.sp,
        letterSpacing = 6.sp
    )
    
    val TitleSmall = TextStyle(
        fontFamily = SerifFamily,
        fontWeight = FontWeight.SemiBold,
        fontSize = 24.sp,
        letterSpacing = 4.sp
    )
    
    val Heading = TextStyle(
        fontFamily = SerifFamily,
        fontWeight = FontWeight.SemiBold,
        fontSize = 20.sp,
        letterSpacing = 2.sp
    )
    
    val Body = TextStyle(
        fontFamily = SerifFamily,
        fontWeight = FontWeight.Normal,
        fontSize = 16.sp,
        lineHeight = 24.sp
    )
    
    val Caption = TextStyle(
        fontFamily = SerifFamily,
        fontWeight = FontWeight.Normal,
        fontSize = 14.sp
    )
    
    val Small = TextStyle(
        fontFamily = SerifFamily,
        fontWeight = FontWeight.Normal,
        fontSize = 12.sp
    )
    
    val CodeLarge = TextStyle(
        fontFamily = FontFamily.Monospace,
        fontWeight = FontWeight.Normal,
        fontSize = 24.sp,
        letterSpacing = 8.sp
    )
}

// ============================================
// MARK: - Custom Shapes
// ============================================
object Parliament1812Shapes {
    // Cut corner button shape
    val ButtonShape = CutCornerShape(
        topStart = 0.dp,
        topEnd = 0.dp,
        bottomStart = 0.dp,
        bottomEnd = 12.dp
    )
    
    // Card shape with slight rounded corners
    val CardShape = androidx.compose.foundation.shape.RoundedCornerShape(12.dp)
    
    // Hexagon shape for badges
    val HexagonShape = GenericShape { size, _ ->
        val centerX = size.width / 2
        val centerY = size.height / 2
        val radius = minOf(size.width, size.height) / 2
        
        moveTo(
            centerX + radius * kotlin.math.cos(-Math.PI / 2).toFloat(),
            centerY + radius * kotlin.math.sin(-Math.PI / 2).toFloat()
        )
        
        for (i in 1..5) {
            val angle = -Math.PI / 2 + i * Math.PI / 3
            lineTo(
                centerX + radius * kotlin.math.cos(angle).toFloat(),
                centerY + radius * kotlin.math.sin(angle).toFloat()
            )
        }
        close()
    }
    
    // Slanted button shapes for mode toggle
    val CreateButtonShape = GenericShape { size, _ ->
        moveTo(size.width * 0.1f, 0f)
        lineTo(size.width, 0f)
        lineTo(size.width * 0.9f, size.height)
        lineTo(0f, size.height)
        close()
    }
    
    val JoinButtonShape = GenericShape { size, _ ->
        moveTo(0f, 0f)
        lineTo(size.width * 0.9f, 0f)
        lineTo(size.width, size.height)
        lineTo(size.width * 0.1f, size.height)
        close()
    }
}

// ============================================
// MARK: - Material3 Color Scheme
// ============================================
private val Parliament1812ColorScheme = darkColorScheme(
    primary = Parliament1812Colors.AntiqueGold,
    onPrimary = Parliament1812Colors.DarkNavy,
    primaryContainer = Parliament1812Colors.MutedGold,
    onPrimaryContainer = Parliament1812Colors.DarkNavy,
    
    secondary = Parliament1812Colors.SaddleBrown,
    onSecondary = Parliament1812Colors.ParchmentCream,
    secondaryContainer = Parliament1812Colors.CardBrown,
    onSecondaryContainer = Parliament1812Colors.ParchmentCream,
    
    tertiary = Parliament1812Colors.ToryBlue,
    onTertiary = Parliament1812Colors.ParchmentCream,
    
    background = Parliament1812Colors.DarkBrown,
    onBackground = Parliament1812Colors.ParchmentCream,
    
    surface = Parliament1812Colors.CardBrown,
    onSurface = Parliament1812Colors.ParchmentCream,
    surfaceVariant = Parliament1812Colors.SecondaryNavy,
    onSurfaceVariant = Parliament1812Colors.MutedText,
    
    error = Parliament1812Colors.NayCrimson,
    onError = Parliament1812Colors.ParchmentCream,
    
    outline = Parliament1812Colors.GoldBorder,
    outlineVariant = Parliament1812Colors.SubtleText,
)

// ============================================
// MARK: - Theme Composable
// ============================================
@Composable
fun Parliament1812Theme(
    content: @Composable () -> Unit
) {
    MaterialTheme(
        colorScheme = Parliament1812ColorScheme,
        typography = Typography(
            displayLarge = Parliament1812Typography.TitleLarge,
            displayMedium = Parliament1812Typography.TitleMedium,
            displaySmall = Parliament1812Typography.TitleSmall,
            headlineLarge = Parliament1812Typography.Heading,
            headlineMedium = Parliament1812Typography.Body,
            bodyLarge = Parliament1812Typography.Body,
            bodyMedium = Parliament1812Typography.Caption,
            bodySmall = Parliament1812Typography.Small,
        ),
        content = content
    )
}

// ============================================
// MARK: - Party Enum
// ============================================
enum class Party(
    val color: Color,
    val nameChinese: String,
    val nameEnglish: String
) {
    TORY(Parliament1812Colors.ToryBlue, "托利黨", "TORY PARTY"),
    WHIG(Parliament1812Colors.WhigOrange, "輝格黨", "WHIG PARTY"),
    NEUTRAL(Parliament1812Colors.Neutral, "中立", "NEUTRAL")
}

// ============================================
// MARK: - Constants
// ============================================
object Parliament1812Dimens {
    // Spacing
    val SpacingXS = 4.dp
    val SpacingS = 8.dp
    val SpacingM = 16.dp
    val SpacingL = 24.dp
    val SpacingXL = 32.dp
    val SpacingXXL = 48.dp
    
    // Padding
    val PaddingButton = 14.dp
    val PaddingCard = 16.dp
    val PaddingScreen = 24.dp
    
    // Borders
    val BorderThin = 1.dp
    val BorderMedium = 2.dp
    val BorderThick = 4.dp
    
    // Corner radius
    val CornerSmall = 4.dp
    val CornerMedium = 8.dp
    val CornerLarge = 12.dp
    val CornerXL = 16.dp
    
    // Button cut corner
    val ButtonCutCorner = 12.dp
    
    // Icon sizes
    val IconSmall = 16.dp
    val IconMedium = 24.dp
    val IconLarge = 32.dp
    val IconXL = 48.dp
    
    // Touch targets
    val MinTouchTarget = 44.dp
}

// ============================================
// MARK: - Shadow & Elevation
// ============================================
object Parliament1812Elevation {
    val Card = 8.dp
    val Button = 4.dp
    val Dialog = 16.dp
    val Overlay = 24.dp
}

// ============================================
// MARK: - Animation Durations
// ============================================
object Parliament1812Animation {
    const val FastMs = 150
    const val NormalMs = 300
    const val SlowMs = 500
    const val VerySlowMs = 800
}

// ============================================
// MARK: - Gradients
// ============================================
object Parliament1812Gradients {
    val GoldButton = listOf(
        Parliament1812Colors.AntiqueGold,
        Parliament1812Colors.MutedGold
    )
    
    val CardBackground = listOf(
        Parliament1812Colors.CardBrown.copy(alpha = 0.95f),
        Parliament1812Colors.CardBrown.copy(alpha = 0.85f)
    )
    
    val Vignette = listOf(
        Color.Transparent,
        Color.Transparent,
        Parliament1812Colors.DarkBrown.copy(alpha = 0.8f)
    )
}

// ============================================
// MARK: - String Resources (Chinese)
// ============================================
object Parliament1812Strings {
    // Home Screen
    const val AppTitle = "1812"
    const val AppSubtitleChinese = "國會風雲"
    const val AppSubtitleEnglish = "Parliament Debates"
    const val RegencyEra = "REGENCY ERA"
    const val BritishParliament = "1812 • British Parliament"
    
    // Mode Toggle
    const val CreateRoom = "建立房間"
    const val JoinRoom = "加入房間"
    const val CreateEnglish = "Create"
    const val JoinEnglish = "Join"
    
    // Input Fields
    const val NicknameChinese = "您的暱稱"
    const val NicknameEnglish = "Your Nickname"
    const val NicknamePlaceholder = "輸入暱稱..."
    const val RoomCodeChinese = "房間代碼"
    const val RoomCodeEnglish = "Room Code"
    const val RoomCodePlaceholder = "XXXXXX"
    
    // Buttons
    const val CreateMeeting = "建立新會議"
    const val EnterChamber = "進入議事廳"
    const val StartGame = "開始遊戲"
    const val Ready = "準備就緒"
    
    // Atmospheric
    const val Tagline = "「在攝政王的注視下，國會的權力鬥爭即將展開」"
    const val TaglineEnglish = "Under the Prince Regent's gaze..."
    
    // Waiting Room
    const val WaitingRoom = "等候大廳"
    const val MembersPresent = "在場成員"
    const val CopyCode = "複製代碼"
    const val Host = "主持"
    const val ReadyStatus = "準備"
    const val Waiting = "等待中"
    
    // Role Reveal
    const val YourRole = "您的身份"
    const val Objective = "您的目標"
    const val Allies = "已知盟友"
    const val HideRole = "隱藏身份"
    const val ShowRole = "顯示身份"
    const val TopSecret = "TOP SECRET"
    
    // Voting
    const val VotingTitle = "議案表決"
    const val Aye = "贊成"
    const val AyeEnglish = "AYE"
    const val Nay = "反對"
    const val NayEnglish = "NAY"
    const val Abstain = "棄權"
    const val AbstainEnglish = "ABSTAIN"
    const val VoteRecorded = "已記錄"
    const val Approved = "APPROVED"
    const val Rejected = "REJECTED"
    
    // Party Names
    const val ToryChinese = "托利黨"
    const val ToryEnglish = "TORY PARTY"
    const val WhigChinese = "輝格黨"
    const val WhigEnglish = "WHIG PARTY"
    const val NeutralChinese = "中立"
    const val NeutralEnglish = "NEUTRAL"
    
    // Errors
    const val EnterNickname = "請輸入您的暱稱"
    const val EnterRoomCode = "請輸入房間代碼"
    const val ConnectionError = "連線錯誤"
    const val RoomNotFound = "找不到房間"
    const val NFCRequired = "請將 NFC 卡片靠近手機"
}
